src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -974,7 +977,12 @@ public IVsEnumWindowSearchOptions SearchOptionsEnum
 
         private void FocusOnSearchBox_Executed(object sender, ExecutedRoutedEventArgs e)
         {
-            _windowSearchHost.Activate();
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
```

Comment:
I wonder if it's necessary for the caller of this method to wait for the UI thread and window activation. If not, then using JTF.RunAsync (with your async package's JTF instance together with FileAndForget) is probably a better approach since it will release background threads more quickly.

Reply:
Caller doesn't need to wait, I just kept it similar to existing flow. But I agree with your assessment that RunAsync will be more efficient approach.